Library/src/main/java/com/pusher/android/notifications/gcm/GCMRegistrationIntentService.java

```diff
@@ -33,38 +27,24 @@ protected void onHandleIntent(Intent intent) {
 
         final InstanceID instanceID = InstanceID.getInstance(this);
         final String defaultSenderId = intent.getStringExtra("gcm_defaultSenderId");
-        String token;
-
-        Callable<String> tokenRetrieval = new Callable<String>() {
-            @Override
-            public String call() throws Exception {
-                return instanceID.getToken(defaultSenderId, GoogleCloudMessaging.INSTANCE_ID_SCOPE, null);
-            }
-        };
-
-        Retryer<String> retryer = RetryerBuilder.<String>newBuilder().
-                retryIfRuntimeException().
-                withStopStrategy(StopStrategies.stopAfterAttempt(INSTANCE_ID_RETRY_ATTEMPTS)).
-                build();
 
         Intent gcmCalled = new Intent(PushNotificationRegistration.GCM_CALLED_INTENT_FILTER);
+        int retryAttempts = 0;
+        String token = null;
+
+        while (token == null && retryAttempts < INSTANCE_ID_RETRY_ATTEMPTS){
+            try {
+                token = instanceID.getToken(defaultSenderId, GoogleCloudMessaging.INSTANCE_ID_SCOPE, null);
+            } catch (IOException e) {
+                retryAttempts++;
```

Comment:
But, what if the `getToken` returns `null`?

Then it will never exit this loop...